src/components/sbb-form-field/sbb-form-field.tsx

```diff
@@ -0,0 +1,213 @@
+import { Component, Host, h, Prop, State } from '@stencil/core';
+import { InterfaceSbbFormFieldAttributes } from './sbb-form-field.custom';
+import { AgnosticMutationObserver as MutationObserver } from '../../global/helpers/mutation-observer';
+
+let nextId = 0;
+
+/**
+ * @slot label - Slot to render a label.
+ * @slot prefix - Slot to render an icon on the left side of the input.
+ * @slot input - Slot to render an input.
+ * @slot suffix - Slot to render an icon on ther right side of the input.
+ * @slot error - Slot to render an error.
+ */
+@Component({
+  shadow: true,
+  styleUrl: './sbb-form-field.scss',
+  tag: 'sbb-form-field',
+})
+export class SbbFormField {
+  /**
+   * Add a specific space if the `<sbb-error>` is present.
+   */
+  @Prop() public errorSpace?: InterfaceSbbFormFieldAttributes['errorSpace'] = 'default';
+
+  /**
+   * Add a `<label>` for the input.
+   */
+  @Prop() public label: string;
+
+  /**
+   * Indicates whether the input is optional.
+   */
+  @Prop() public optional?: boolean;
+
+  /**
+   * Size variant, either l or m.
+   */
+  @Prop() public size?: InterfaceSbbFormFieldAttributes['size'] = 'l';
+
+  /** Whether to display the form field without a border. */
+  @Prop() public borderless = false;
+
+  /**
+   *
+   * It is used internally to get the native `disabled` attribute from `<input>`.
+   */
+  @State() private _disabled: boolean;
+
+  /**
+   *
+   * It is used internally to get the native `readonly` attribute from `<input>`.
+   */
+  @State() private _readonly: boolean;
+
+  /**
+   * It is used internally to get the `prefix` slot.
+   */
+  @State() private _prefixElements: Element[] = [];
+
+  /**
+   * It is used internally to get the `suffix` slot.
+   */
+  @State() private _suffixElements: Element[] = [];
+
+  /**
+   * It is used internally to get the `suffix` slot.
+   */
+  @State() private _errorElements: Element[] = [];
+
+  /** Whether the input inside the form field is invalid. */
+  @State() private _invalid = false;
+
+  private _originalInputAriaDescribedby?: string;
+
+  /**
+   * It is used internally to get the `input` slot.
+   */
+  private _input: HTMLInputElement | HTMLSelectElement | HTMLElement;
```

Comment:
Can't it be theoretically be undefined?

Reply:
It's a good point; what should we do if consumers don't provide the <input> ? 
- should we trigger a console.warn(..) as in the sbb-tab-group?
- should we renderer something as in https://github.com/lyne-design-system/lyne-components/blob/4a4a738dd72d3bbc5d9ce506631dc6f86685619e/src/components/sbb-button/sbb-button.tsx#L78 ?

Reply:
@kyubisation what do you think?

Reply:
Discussed with @jeripeierSBB; We do not provide a warning for now. Since this is technically an integral part of the component, we assume (for now) that a consumer will notice this themselves.